Feature/task validation

slava225678 opened the pull request:
Тикет  PR:
    ___#168___

slava225678 committed:
Временный коммит для переключения веток

slava225678 committed:
Временный коммит для переключения веток

slava225678 committed:
тестировочный коммит для восстановления миграций

slava225678 committed:
пустой коммит, хочу слить изменения из dev

slava225678 committed:
Merge branch 'dev' into feature/task-validation

slava225678 committed:
проведена работа над валидаторами

slava225678 committed:
Update dev.txt

slava225678 committed:
Update prod.txt

DinarM committed:
add new workflow pytest (#205)

AndreyZherdetskiy commented on src/api/v1/validators/tasks_validators.py:
```diff
@@ -0,0 +1,45 @@
+    Raises:
+        HTTPException: Если задача не найдена.
+    """
+    result = await session.execute(select(Task).where(Task.id == task_id))
```
Думаю, что такой проверки тут достаточно
return await task_crud.get_or_404(session, task_id, message=ERROR_TASK_NOT_FOUND)

AndreyZherdetskiy commented on src/api/v1/validators/tasks_validators.py:
```diff
@@ -0,0 +1,45 @@
+        )
+
+
+async def check_tasks_exists(company_slug: str, problem_id: int, session: AsyncSession):
```
Давай такое название, чтобы не было путаницы
check_tasks_for_company_problem_exist

slava225678 replied:
Сорри, сам что-то профукал момент)

AndreyZherdetskiy commented on src/problems/validators/task_validators.py:
```diff
@@ -0,0 +1,66 @@
+)
+
+
+def validate_date_in_future(value: date) -> date:
```
Давай так:
```
def validate_date_in_future(value: date | None) -> date | None:
    """Валидирует дату встречи.

    Назначение:
        Валидирует, что строка:
        1. Не находится в прошлом
    Параметры:
        value: Дата для валидации.
    Возвращает:
        Проверенная дата.
    Исключения:
        ValueError: Если дата в прошлом.
    """
    if value is None:
        return None
    if value < date.today():
        raise ValueError(f'{ERROR_DATE_SHOULD_BE_FUTURE} {date.today()}')
    return value
```
В схемах просто вызов функции даже для опциональных полей

AndreiAgeev committed:
Bugfix/problem feeds comment validation (#203)

В процессе написания тестов нашил одну проблему: при запросах к ручкам
редактирования/удаления комментариев не происходила сверка
**message_feed**, к которому относится комментарий и
**message_feed_id**, который передавался в запросе. Т.е. можно было
изменить/удалить комментарий, если в запросе указать неверный
**message_feed_id**, но корректный **comment_id**. Сейчас это исправлено

slava225678 committed:
Feature/task-validation(204)

Тикет  PR:
    ___#168___

slava225678 committed:
Merge branch 'dev' into feature/task-validation

slava225678 committed:
разрешал конфликты, поправил ручку и круд patch запроса

slava225678 committed:
Resolve merge conflicts

slava225678 committed:
Merge branch 'dev' into feature/task-validation

slava225678 committed:
Merge branch 'dev' into feature/task-validation

AndreyZherdetskiy reviewed (commented):
Блин, забыл закрыть ревью по ходу дела

slava225678 committed:
исправление рекомендаций и замечаний ревьюера

slava225678 committed:
Feature/task-validation(204)

Тикет  PR:
    ___#168___

slava225678 committed:
исправление рекомендаций и замечаний ревьюера

AndreyZherdetskiy commented on src/problems/crud/task_crud.py:
```diff
@@ -96,7 +96,7 @@ async def get_task_by_id(
         )
         result = await session.execute(query)
-        task = result.scalar_one_or_none()
+        task = result.scalar()
```
А что тут поменялось? Ты же также должен получить уникальную таску. Обрати внимание на это момент еще раз, пожалуйста

AndreyZherdetskiy commented on src/api/v1/endpoints/tasks.py:
```diff
@@ -106,10 +114,10 @@ async def get_task(
+    await check_company_exists(company_slug, session)
+    await check_task_exists(task_id, session)
+    await check_problem_exists(problem_id, session)
+    return await task_crud.get_task_by_id(session, company_slug, problem_id, task_id)  # type: ignore
```
Тут и далее в ПР убери, пожалуйста, # type: ignore

AndreyZherdetskiy commented on src/problems/crud/task_crud.py:
```diff
@@ -159,30 +159,39 @@ async def create(
             old_date_completion = db_obj.date_completion
             update_data = obj_in.model_dump(exclude_unset=True)
             executors_data = update_data.pop('executors', None)
+            # Обновляем поля задачи
```
Такие комменты тоже почисти

slava225678 committed:
исправление рекомендаций и замечаний ревьюера
